libsolidity/ast/Types.h

```diff
@@ -208,14 +208,30 @@ class Type
 	virtual bool operator==(Type const& _other) const { return category() == _other.category(); }
 	virtual bool operator!=(Type const& _other) const { return !this->operator ==(_other); }
 
-	/// @returns number of bytes used by this type when encoded for CALL. If it is a dynamic type,
-	/// returns the size of the pointer (usually 32). Returns 0 if the type cannot be encoded
-	/// in calldata.
+	/// @returns number of bytes used by this type when encoded for CALL. Cannot be used for
+	/// dynamically encoded types.
+	/// Always returns a value greater than zero and throws if the type cannot be encoded in calldata
+	/// (or is dynamically encoded).
 	/// If @a _padded then it is assumed that each element is padded to a multiple of 32 bytes.
-	virtual unsigned calldataEncodedSize(bool _padded) const { (void)_padded; return 0; }
+	virtual unsigned calldataEncodedSize(bool _padded) const { (void)_padded; solAssert(false, ""); }
+	/// @returns the distance between two elements of this type in an array, tuple or struct.
+	/// For statically-sized types this is the same as calldataEncodedSize(true).
+	/// For dynamically-encoded types this is the distance between two tail pointers, i.e. 32.
+	/// Always returns a value greater than zero and throws if the type cannot be encoded in calldata.
+	virtual unsigned calldataHeadIncrement() const { return calldataEncodedSize(); }
```

Comment:
Does it make sense to make this non-virtual and implement it exactly as it is described above?

Reply:
As a matter of fact it does, yes - I hadn't realized that was possible - I'll change it!

Reply:
Why not call this `calldataHeadSize` to make it consistent with `memoryHeadSize` - is there a big difference?

Reply:
My PR comment above already suggested we could also name it ``calldataHeadSize`` - I'm fine with that if you prefer! It's currently called increment, because incrementing head pointers is the main thing it's used for, but ``calldataHeadSize`` is fine with me as well.

Reply:
Renamed it to ``calldataHeadSize`` in the latest commit.